src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpoint.java

```diff
@@ -182,10 +185,35 @@ DockerRegistryToken getToken(Item context) {
 
         // look for subtypes that know how to create a token, such as Google Container Registry
         return AuthenticationTokens.convert(DockerRegistryToken.class, firstOrNull(CredentialsProvider.lookupCredentials(
-                IdCredentials.class, context, Jenkins.getAuthentication(),requirements),
+                IdCredentials.class, context, Tasks.getAuthenticationOf((Queue.Task)context),requirements),
```

Comment:
Smells wrong. As you can see from the Javadoc of that method, it is a _prediction_, not an indication of what _actual_ authentication is associated with the build—which would be `Jenkins.getAuthentication()`.

Reply:
This is an attempt to fix the JENKINS-48437 without needing to update the dependencies in `docker-worflow` (or other dependents of `docker-commons`).